hybridse/src/udf/udf.cc

```diff
@@ -64,6 +64,28 @@ void hex(StringRef *str, StringRef *output) {
     output->data_ = buffer;
 }
 
+void unhex(StringRef *str, StringRef *output) {
+    std::ostringstream ss;
+    uint32_t arr[str->size_];
+    for (uint32_t i=0; i < str->size_; i++) {
+        uint32_t tmp;  
+        if (str->data_[i] <= 'F' && str->data_[i] >= 'A') {
+            tmp = str->data_[i] - 'A' + 10;
+        } else if (str->data_[i] <= 'z' && str->data_[i] >= 'a') {
+            tmp = str->data_[i] - 'a' + 10;
+        } else if (str->data_[i] <= '9' && str->data_[i] >= '0') {
+            tmp = str->data_[i] - '0';
+        }
+        arr[i] = tmp;
+    }
+    for (uint32_t i=0; i < str->size_; i+=2) {
```

Comment:
two level inner array is used, which can be avoided. 

I think you can take first two characters from input and parse it as hex number. Then write into output directly, since output size is known from input size.

Reply:
will fix that soon, thanks for the review :)